src/server.cpp

```diff
@@ -251,8 +251,14 @@ Server::Server(
 
 	// Apply texture overrides from texturepack/override.txt
 	std::string texture_path = g_settings->get("texture_path");
-	if (!texture_path.empty() && fs::IsDir(texture_path))
-		m_nodedef->applyTextureOverrides(texture_path + DIR_DELIM + "override.txt");
+	if (!texture_path.empty() && fs::IsDir(texture_path)) {
+		std::vector<std::string> paths;
+		paths.push_back(texture_path);
+		std::set<char> chars_to_ignore = { '_', '.' };
+		fs::GetRecursiveSubPaths(texture_path, paths, false, &chars_to_ignore);;
```

Comment:
1) spotted `;;`
2) Could you please move the block, start=`std::string texture_path`, end=`fs::GetRecursiveSubPaths` into a function? An empty vector can be returned if the path is not set or not a directory.